node/pkg/driver/device_connectivity/device_connectivity_helper_scsigeneric.go

```diff
@@ -506,48 +588,60 @@ func (o OsDeviceConnectivityHelperGeneric) ReloadMultipath() error {
 	return nil
 }
 
-func (o OsDeviceConnectivityHelperGeneric) GetDmsPath(volumeId string, volumeNguid string) (string, error) {
-	volumeUuidLower := strings.ToLower(volumeId)
-
-	mpathdOutput, err := o.Helper.WaitForDmToExist(volumeUuidLower, volumeNguid, WaitForMpathRetries, WaitForMpathWaitIntervalSec)
-
+func (o OsDeviceConnectivityHelperGeneric) GetMpathdOutputByVolumeId(volumeIdVariations []string) (string, error) {
+	mpathdOutput, err := o.Helper.GetMpathdOutput(volumeIdVariations, multipathdWildcardsMpathNameAndVolumeId)
 	if err != nil {
 		return "", err
 	}
+	return mpathdOutput, nil
+}
 
-	if mpathdOutput == "" {
-		return "", &MultipathDeviceNotFoundForVolumeError{volumeId}
+func (o OsDeviceConnectivityHelperGeneric) GetVolumeIdByMpathdName(mpathDeviceName string, mpathdOutput string) (string, error) {
+	volumeIdVariations := []string{mpathDeviceName}
```

Comment:
`mpathDeviceName` is not a volume id variation

Reply:
done